### PR TITLE
Allow to load client certificate when connecting to RabbitMQ server

### PR DESCRIPTION
**Pull request by bvis**
- Update AMQP and Elasticsearch libraries
- Allow to load client certificate when connecting to RabbitMQ server

Signed-off-by: Basilio Vera <basilio.vera@softonic.com>

### CODE REVIEW

File: pkg/services/container.go [hunk 1]
```diff
@@ -64,13 +64,20 @@ var Container = []dingo.Def{
 					cfg.RootCAs.AppendCertsFromPEM(ca)
 					klog.V(4).Infof("Added CA certificate %s", caPath)
 				}
+                tlsClientCert := os.Getenv("RABBITMQ_TLS_CLIENT_CERT")
+                tlsClientKey := os.Getenv("RABBITMQ_TLS_CLIENT_KEY")
+
+                if tlsClientCert != "" {
+                    if cert, err := tls.LoadX509KeyPair(tlsClientCert, tlsClientKey); err == nil {
```
Comment (rccrdpccl) [inline pkg/services/container.go:71]: If this errors, a connection will be tried anyway, right?

**Review by rccrdpccl — CHANGES_REQUESTED** [review]
Please use `gofmt` to format the code

File: pkg/services/container.go [hunk 2]
```diff
@@ -64,13 +64,20 @@ var Container = []dingo.Def{
 					cfg.RootCAs.AppendCertsFromPEM(ca)
 					klog.V(4).Infof("Added CA certificate %s", caPath)
 				}
+                tlsClientCert := os.Getenv("RABBITMQ_TLS_CLIENT_CERT")
+                tlsClientKey := os.Getenv("RABBITMQ_TLS_CLIENT_KEY")
+
+                if tlsClientCert != "" {
```
[the remark this answers is not in the document]
Comment (bvis) [reply pkg/services/container.go:70]: I guessed that they come in pair, probably it's better to check both

File: pkg/services/container.go (hunk 1 above)
Comment (bvis) [reply pkg/services/container.go:71]: yes

Comment (rccrdpccl) [reply pkg/services/container.go:71]: Should we notice the user about it?

Comment (bvis) [reply pkg/services/container.go:71]: Do you mean to add a notice to the logs?

**Comment by bvis** [comment]
Please check https://github.com/softonic/homing-pigeon/pull/11 instead

**Pull request closed without merging**